### ISSUE TITLE
Add the Real Server Load Event

### ISSUE THREAD

**Issue by erenkarakal** [SkriptLang/Skript#5558]
### Suggestion

The current [""server load"" event](https://skripthub.net/docs/?id=995) is called when Skript starts, but there is an event for actual [server load](https://jd.papermc.io/paper/1.19/org/bukkit/event/server/ServerLoadEvent.html). That should be used instead.

### Why?

The current syntax is misleading. (even though it throws a warning)

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this suggestion.

**Comment by AyhamAl-Ali** [comment]
I think the only difference here is that there is an actual knowledge when the server actually **starts up** or **reloads**, the current Skript implementation doesn't differentiate between those and trigger the event if Skript plugin was reloaded using /reload or via a plugin like PlugMan, this suggestion is nice to have.

**Comment by TheLimeGlass** [comment]
on skript load and on server start at the same event because Skript only knows when it starts itself, not the actual server as it's a plugin. If the plugin gets disabled, Skript considers that to be a server shutdown as.Skript cannot run without being a plugin.

**Comment by bluelhf** [comment]
> on skript load and on server start are the same event because Skript only knows when it starts itself, not the actual server as it's a plugin. If the plugin gets disabled, Skript considers that to be a server shutdown as Skript cannot run without being a plugin. That's where the warning reasoning reflects this.

Yes, that's the behaviour this issue is proposing be changed. The server load event would change it by differentiating between reloads, server startups and plugin loads (with something like PlugMan, which I'm pretty sure breaks Skript anyway).

With the new behaviour,
- The server start event would run only after the server is completely done loading, instead of when Skript first enables.
- Scripts could differentiate between server startup and server reload.
- Reloading Skript programmatically wouldn't fire the `on server start` event.

This would better reflect the syntax and expected behaviour of the event.

**Comment by AyhamAl-Ali** [comment]
> This sounds like it would be a breaking change for people using the event in its current form.

It will be but not that much, people  using `on server start` will already have issues (probably) when they try to reload Skript using /reload or PlugMan, so it's kinda just more stable event after this enhancement I would say.

I would suggest the new syntaxes to be 
`on (server|skript) (start|reload)[s]`

Something like this to keep the current feature of skript starting/reloading event (for possible reloads -- if they even work), and adding server starting/reloading event

**Comment by erenkarakal** [comment]
> on skript load and on server start are the same event because Skript only knows when it starts itself, not the actual server as it's a plugin. If the plugin gets disabled, Skript considers that to be a server shutdown as Skript cannot run without being a plugin. That's where the warning reasoning reflects this.

I meant an event that fires after the server is fully loaded (all plugins are loaded and the server is ready to join) I think the one I sent isn't it. Perhaps there is another way?
